api-platform/core
# feat(laravel): laravel component

# Laravel support in API Platform

Note that this is still in development there are a few things missing (see todolist)

Must do:

- [x] Eloquent pagination
- [ ] Eloquent filters
- [x] Add a TestBundle inside `src/Laravel` to create functional tests
- [x] Eloquent ApiProperties (cf https://github.com/api-platform/core/issues/5885)
- [x] integrate Eloquent relations
- [x] take ideas from https://github.com/api-platform/core/pull/4197 (relation persistence ? should we integrate eloquent bundle ?)
- [x]  JSON:API normalizers
- [ ]  ResourceAccessChecker

Some ideas: 
- [ ] providers/processors as middlewares
- [ ] add the graphQL support? Not a priority yet as lighthouse-php is generally used, we may see in the future how we could integrate with lighthouse 

To try this:

1. create a laravel project:

```
composer create-project laravel/laravel my-api
```

2. update your composer.json with:

```
    "repositories": [
        {
            "type": "path",
            "url": "/path/to/core/src/Laravel"
        }
    ]
}
```

3. `composer update`, I added `#[ApiResource]` on the model.
